server/server.js

```diff
@@ -81,4 +74,36 @@ app.post('/api/v1/shoes/:id/comments', async (request, response) => {
     }
 });
 
+// post new shoe
+app.post('/api/v1/shoes', async (request, response) => {
+    const shoe = request.body;
+
+    for(let requiredParam of ['brand', 'colorway', 'retail_price', 'model']) {
+        if(!shoe[requiredParam]) {
+            return response
+                .status(422)
+                .send({ error: `Expected format: { brand: <String>, colorway: <String>, retail_price: <String>, model: <String> }. You're missing a "${requiredParam}" property.` });  
+        }
+    }
+    try {
+     const shoeData = await database('shoes').insert(shoe, ['id', 'thumb_url', 'title', 'product', 'brand', 'colorway', 'demographic', 'retail_price', 'model', 'year', 'small_image_url', 'image_url'])
+     const fromDatabase = shoeData[0] 
+     shoe.id = fromDatabase.id
```

Comment:
great re assignment of variables here to get back proper data from the postrequest.